### PR TITLE
simple gmres

### PR DESCRIPTION
**Pull request by bmcdanie**
Please review the [developer documentation](https://github.com/project-asgard/asgard/wiki/developing)
on the wiki of this project that contains help and requirements.

## Proposed changes

1) add simple iterative solver that works locally with constructed matrix.
2) add distributed iterative solver that works across nodes with matrix shell (GEMV)

## What type(s) of changes does this code introduce?
_Put an `x` in the boxes that apply._

- [ ] Bugfix
- [x] New feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### Does this introduce a breaking change?

- [ ] Yes
- [x] No

## What systems has this change been tested on?

None yet

## Checklist

_Put an x in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- [ ] this PR is up to date with current the current state of 'develop'
- [x] code added or changed in the PR has been clang-formatted
- [ ] this PR adds tests to cover any new code, or to catch a bug that is being fixed
- [ ] documentation has been added (if appropriate)

### CODE REVIEW

**Review by stonecoldhughes — APPROVED** [review]
send it

**Review by stonecoldhughes — DISMISSED** [review]
disapproved

File: src/solver.cpp [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+      {
+        fm::getrs(precond, new_basis, precond_pivots);
+      }
+
+      for (int k = 0; k <= i; ++k)
+      {
+        fk::vector<P, mem_type::const_view> const basis_vect(basis, k, 0,
+                                                             basis.nrows() - 1);
+        krylov_proj(k, i) = new_basis * basis_vect;
+        krylov_proj.print("krylov");
```
Comment (stonecoldhughes) [inline src/solver.cpp]: don't forget to take out the debug prints in the final version

**Review by stonecoldhughes — CHANGES_REQUESTED** [review]

File: src/solver.cpp (hunk 1 above)
Comment (bmcdanie) [reply src/solver.cpp]: good catch, done

**Pull request merged**